swiftlint: new submission

csknns opened the pull request:
#### Description

A new portfile for [https://github.com/realm/SwiftLint swiftlint], a simple command line tool to enforce Swift style and and conventions.

###### Tested on
macOS 13.5.2 22G91 arm64
Xcode 14.3.1 14E300c

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [x] checked your Portfile with `port lint --nitpick`?
- [x] tried existing tests with `sudo port test`?
- [x] tried a full install with `sudo port -vst install`?
- [x] tested basic functionality of all binary files?
- [x] checked that the Portfile's most important [variants](https://trac.macports.org/wiki/Variants) haven't been broken?

herbygillot commented on devel/swiftlint/Portfile:
```diff
@@ -0,0 +1,41 @@
+
+github.setup        realm swiftlint 0.52.4
+
+name                swiftlint
```
`github.setup` will implicitly set `name` & `version`, so you can drop this and the following lines.

herbygillot commented on devel/swiftlint/Portfile:
```diff
@@ -0,0 +1,41 @@
+name                swiftlint
+version             0.52.4
+categories          devel
+platforms           darwin
```
`platforms darwin` is implicit now, and no longer need to be specified.

herbygillot commented on devel/swiftlint/Portfile:
```diff
@@ -0,0 +1,41 @@
+categories          devel
+platforms           darwin
+license             MIT
+maintainers         gmail.com:christos.koninis
```
A port should also be marked `openmaintainer` unless there is a very specific reason why you and only you should update or make any changes to the port:

```suggestion
maintainers         gmail.com:christos.koninis \
                    openmaintainer
```

reneeotten replied:
and please add your GitHub handle

csknns replied:
@reneeotten Nice I missed that about the maintainers syntax. I added my Github handle

herbygillot commented on devel/swiftlint/Portfile:
```diff
@@ -0,0 +1,41 @@
+use_configure       no
+use_xcode           yes
+
+build.cmd           /usr/bin/swift
```
Choose one of `swift` or `/usr/bin/swift` and ideally use the same command for both `build.cmd` and `test.cmd`?

csknns replied:
You are right they should both be the same I think plain swift is OK no need for absolut path.

herbygillot commented:
Thank you for this contribution, @csknns.  Some comments above.

Also, the commit message needs to be in the format `<portname>: <description>`.  So in this case, it should look like:

`swiftlint: new port`

reneeotten commented on devel/swiftlint/Portfile:
```diff
@@ -0,0 +1,41 @@
+PortGroup           xcodeversion 1.0
+
+github.setup        realm swiftlint 0.52.4
+
```
please add the "github.tarball_from" line and set it to "releases"(if upstream provides those) or use "archive". Making this change will require you to update the "checksums".

reneeotten replied:
please add the "revision 0" line

csknns commented:
@herbygillot @reneeotten Thank you for your comments, since this is my first portfile and I am still learning they are very helpful! 
I have updated the PR to fix the issues. I also squashed the commits as per commit guidelines.

mascguy commented:
It looks like CI is failing for Big Sur, with the following:

```
.../swiftlint/work/SwiftLint-0.52.4: error: package at '.../swiftlint/work/SwiftLint-0.52.4'
is using Swift tools version 5.7.0 but the installed version is 5.5.0
```

Does this port require Swift 5.7 and later?

mascguy commented:
> It looks like CI is failing for Big Sur, with the following:
> 
> ```
> .../swiftlint/work/SwiftLint-0.52.4: error: package at '.../swiftlint/work/SwiftLint-0.52.4'
> is using Swift tools version 5.7.0 but the installed version is 5.5.0
> ```
> 
> Does this port require Swift 5.7 and later?

If so, then no worries. But you'll want to add the following, to declare that it's only supported on Monterey and later:

`platforms {darwin >= 12}`

That should be added toward the top of the portfile, such as the line prior to `categories xxx`.

mascguy commented:
Sorry, my previous comment had a typo. For Monterey and later, the declaration should be:

> `platforms {darwin >= 21}`

herbygillot commented on devel/swiftlint/Portfile:
```diff
@@ -0,0 +1,42 @@
+PortGroup           github 1.0
+PortGroup           xcodeversion 1.0
+
+revision 0
```
Can you move this underneath `github.tarball`?

```
github.setup        realm swiftlint 0.52.4
github.tarball_from archive
revision            0
```

csknns replied:
Moved it!

herbygillot commented:
Thanks for making these changes, @csknns. Other than @mascguy's comments, I had a few more minor ones as well.

csknns commented:
> > It looks like CI is failing for Big Sur, with the following:
> > ```
> > .../swiftlint/work/SwiftLint-0.52.4: error: package at '.../swiftlint/work/SwiftLint-0.52.4'
> > is using Swift tools version 5.7.0 but the installed version is 5.5.0
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > Does this port require Swift 5.7 and later?
> 
> If so, then no worries. But you'll want to add the following, to declare that it's only supported on Monterey and later:
> 
> `platforms {darwin >= 21}`
> 
> That should be added toward the top of the portfile, such as the line prior to `categories xxx`.

I added the `platforms {darwin >= 21}` since it requires Swift 5.7 and later. It looks ok know I think.

csknns commented on devel/swiftlint/Portfile:
```diff
@@ -0,0 +1,42 @@
+
+categories          devel
+license             MIT
+maintainers         gmail.com:christos.koninis \
```
I added it.

mascguy commented:
> I added the `platforms {darwin >= 21}` since it requires Swift 5.7 and later. It looks ok know I think.

It looks like you might have copied my original (incorrect) comment, as you added: `platforms {darwin >= 12}`

mascguy commented:
> > I added the `platforms {darwin >= 21}` since it requires Swift 5.7 and later. It looks ok know I think.
> 
> It looks like you might have copied my original (incorrect) comment, as you added: `platforms {darwin >= 12}`

Fixed that for you, and looks good now. Thank you so much for your contribution!
